File: rusi_trader_mobile_new/lib/models/recommendation_model.dart
class RecommendationModel {
  final String recommendation;

  final double confidence;

  final double score;

  final String optionSymbol;

  final double entryPrice;

  final double stopLoss;

  final double targetPrice;

  RecommendationModel({
    required this.recommendation,
    required this.confidence,
    required this.score,
    required this.optionSymbol,
    required this.entryPrice,
    required this.stopLoss,
    required this.targetPrice,
  });

  factory RecommendationModel.fromJson(
      Map<String, dynamic> json) {
    return RecommendationModel(
      recommendation:
          json["recommendation"]?.toString() ?? "--",

      confidence:
          (json["confidence"] as num?)?.toDouble() ?? 0.0,

      score:
          (json["score"] as num?)?.toDouble() ?? 0.0,

      optionSymbol:
          json["option_symbol"]?.toString() ?? "--",

      entryPrice:
          (json["entry_price"] as num?)?.toDouble() ?? 0.0,

      stopLoss:
          (json["stop_loss"] as num?)?.toDouble() ?? 0.0,

      targetPrice:
          (json["target_price"] as num?)?.toDouble() ?? 0.0,
    );
  }
}

File: backend/models/recommendation_model.py
"""
============================================================

Recommendation Model

============================================================
"""

from pydantic import BaseModel


class RecommendationModel(BaseModel):

    recommendation: str | None

    confidence: float | None

    score: float | None

    option_symbol: str | None

    exchange: str | None

    strike: float | None

    expiry: str | None

    option_type: str | None

    entry_price: float | None

    stop_loss: float | None

    target_price: float | None

    risk_reward: float | None

    reasons: list[str]

    updated_time: str
